xmake/modules/core/tools/gdc.lua

```diff
@@ -11,15 +11,162 @@
 -- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
--- 
+--
 -- Copyright (C) 2015-2020, TBOOX Open Source Group.
 --
--- @author      ruki
+-- @author      ruki, BarrOff
 -- @file        gdc.lua
 --
 
--- inherit dmd
-inherit("dmd")
+-- imports
+import("core.base.option")
+import("core.project.config")
+import("core.project.project")
+
+-- init it
+function init(self)
+    
+    -- init arflags
+    self:set("dcarflags", "-lib")
+
+    -- init shflags
+    self:set("dcshflags", "-shared", "-fPIC")
+
+    -- init dcflags for the kind: shared
+    self:set("shared.dcflags", "-fPIC")
+end
+
+-- make the optimize flag
+function nf_optimize(self, level)
+    local maps =
+    {
+        fast        = "-O"
+    ,   faster      = "-O -frelease"
+    ,   fastest     = "-O -frelease -fbounds-check=off"
+    ,   smallest    = "-O -frelease -fbounds-check=off"
+    ,   aggressive  = "-O -frelease -fbounds-check=off"
+    }
+    return maps[level]
+end
```

Comment:
Most of the settings of ldc2/gdc are the same, you can inherit the configuration of dmd and only rewrite the configuration interface that needs to be modified. for example 

```lua
inherit("dmd")

function nf_optimize(self, level)
    local maps =
    {
        fast        = "-O"
    ,   faster      = "-O -frelease"
    ,   fastest     = "-O -frelease -fbounds-check=off"
    ,   smallest    = "-O -frelease -fbounds-check=off"
    ,   aggressive  = "-O -frelease -fbounds-check=off"
    }
    return maps[level]
end

function nf_includedir(self, dir)
    return "-I" .. os.args(dir)
end

-- make the link flag
function nf_link(self, lib)
    return "-l" .. lib
end
```